tmt/steps/prepare/artifact/providers/koji.py

```diff
@@ -87,22 +87,32 @@ class KojiArtifactProvider(ArtifactProvider[RpmArtifactInfo]):
     Currently only supports RPM artifacts.
 
     Example:
-        provider = KojiArtifactProvider(logger, "koji.build:123456")
-        artifacts = provider.download_artifacts(guest, Path("/tmp"), [] )
+        provider = KojiArtifactProvider(logger, build_id=123456)
+        provider = KojiArtifactProvider(logger, task_id=654321)
+        provider = KojiArtifactProvider(logger, nvr="tmt-1.58.0.dev21+gb229884df-main.fc41.noarch")
+        artifacts = provider.download_artifacts(guest, Path("/tmp"), [])
     """
 
     API_URL = "https://koji.fedoraproject.org/kojihub"  # For metadata
 
-    def __init__(self, logger: tmt.log.Logger, artifact_id: str):
-        super().__init__(logger, artifact_id)
+    def __init__(
+        self,
+        logger: tmt.log.Logger,
+        *,
+        build_id: Optional[int] = None,
+        task_id: Optional[int] = None,
+        nvr: Optional[str] = None,
+    ):
+        super().__init__(logger)
         self._session = self._initialize_session()
+        self.artifact_id = self._resolve_artifact_id(build_id, task_id, nvr)
```

Comment:
I would say not to save `self.artifact_id` because it is ambiguous after that what does it represent and how to get the RPMs. Either enforce we are always navigating to the `task_id` or save all 3 and do if checks in the implementations depending on what was provided.